src/reconcile/rec_write.c

```diff
@@ -1290,8 +1309,8 @@ __rec_txn_read(WT_SESSION_IMPL *session, WT_RECONCILE *r,
 		}
```

Comment:
I think we have the same problem in the checkpoint loop we have in the eviction loop: `__wt_txn_upd_visible` returns true and then `upd->txnid` is updated, leading to assertion failure at line 1312.

This one is a little trickier because `__wt_txn_upd_visible` takes a `upd` argument, again potentially leading to an invented load.